vault/core.go

```diff
@@ -2224,6 +2224,10 @@ func (c *Core) sealInternalWithOptions(grabStateLock, keepHALock, performCleanup
 		if keepHALock {
 			atomic.StoreUint32(c.keepHALockOnStepDown, 1)
 		}
+		if grabStateLock {
+			cancelCtxAndLock()
+			defer c.stateLock.Unlock()
```

Comment:
My memory was that this caused deadlocks in the shutdown process, but I wasn't sure _why_. I think it happened on standby nodes? 🤔 

But I would find that surprising. Let me see what #2016 yields.